lib/elixir/src/elixir_fn.erl

```diff
@@ -128,12 +128,12 @@ validate(Meta, [{Pos, _} | _], Expected, E) ->
 validate(_Meta, [], _Pos, _E) ->
   [].
 
-escape({'&', _, [Pos]}, _E, Dict) when is_integer(Pos), Pos > 0 ->
+escape({'&', Meta, [Pos]}, _E, Dict) when is_integer(Pos), Pos > 0 ->
   % Using a nil context here to emit warnings when variable is unused.
   % This might pollute user space but is unlikely because variables
   % named :"&1" are not valid syntax.
-  Var = {list_to_atom([$& | integer_to_list(Pos)]), [], nil},
-  {Var, orddict:store(Pos, Var, Dict)};
+  Var = {list_to_atom([$& | integer_to_list(Pos)]), Meta, nil},
+  {Var, orddict:store(Pos, setelement(2, Var, []), Dict)};
```

Comment:
I didn't add meta on the left arg of `->` as that node is not present in the code

Reply:
I would keep it on both sides, the more information, potentially the better?